modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java

```diff
@@ -148,18 +171,68 @@ public Map<String, PropertyDefinition> getExtendedPropertyDefinitions(
 						"Information about the object relationship ",
 						objectRelationship.getName(),
 						" can be embedded with \"nestedFields\"."),
-					objectRelationship.getName(),
-					PropertyDefinition.PropertyType.MULTIPLE_ELEMENT,
-					new UnsupportedOperationPropertyValidator(), false));
+					objectRelationship.getName(), propertyType,
+					propertyValidator, false));
 		}
 
 		return extendedPropertyDefinitions;
 	}
 
 	@Override
 	public void setExtendedProperties(
-		long companyId, long userId, String className, Object entity,
-		Map<String, Serializable> extendedProperties) {
+			long companyId, long userId, String className, Object entity,
+			Map<String, Serializable> extendedProperties)
+		throws Exception {
+
+		ObjectDefinition objectDefinition = getObjectDefinition(
+			companyId, className);
+
+		if (objectDefinition == null) {
+			throw new BadRequestException(
```

Comment:
Not sure if we should throw this exception here for two reasons:

- If the objectDefinition value is null, is that a client error? I think it is more a server error as it is not expected to be null
- Maybe we should return a more specific exception (not a JAX-RS one) and the ExceptionMapper will map if it applies